tests/pingpong.rs

```diff
@@ -136,3 +145,117 @@ fn udp_ping_pong() {
         }
     }
 }
+
+//==============================================================================
+// TCP Ping Pong
+//==============================================================================
+
+#[test]
+fn tcp_ping_pong() {
+    let mut test: Test = Test::new();
+    let fill_char: u8 = 'a' as u8;
+    let nrounds: usize = 1024;
+    let local_addr: Ipv4Endpoint = test.local_addr();
+    let remote_addr: Ipv4Endpoint = test.remote_addr();
+    let expectbuf: Vec<u8> = test.mkbuf(fill_char);
+
+    // Setup peer.
+    let sockqd: QDesc = test
+        .libos
+        .socket(libc::AF_INET, libc::SOCK_STREAM, 0)
+        .unwrap();
```

Comment:
I don't think we should ever be using unwrap().  The only exception would be if it is a logic error for the underlying routine to return the None or Err case instead of the Some or Ok case.  For test code running outside of a test framework, I think using expect() might be okay (if the error message was good).  But inside a test framework that is expecting explicit failures, we should check for error returns and explicitly fail so the test framework does the right thing.  And of course in production code we shouldn't use unwrap() or expect() as we need to always handle errors gracefully.

This comment applies to each instance of unwrap in both of these test files.

Reply:
Sure. These passed unoticed.

Reply:
Thanks for fixing these, looks much better now.